README.md

```diff
@@ -1 +1,287 @@
+```
+ballerina-gke-deployment
+ └── guide
+      └── hello_world_service.bal
+```
+
+- Create the above directories in your local machine and also create empty `.bal` file.
+
+- Then open the terminal and navigate to `ballerina-gke-deployment/guide` and run Ballerina project initializing toolkit.
+```bash
+   $ ballerina init
+```
+
+As the first step, add the following content to the hello_world_service.bal.
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+import ballerinax/kubernetes;
+
+listener http:Listener httpListener = new(9090);
+
+// By default, Ballerina exposes a service via HTTP/1.1.
+service hello on httpListener {
+
+    // Invoke all resources with arguments of server connector and request.
+    resource function sayHello(http:Caller caller, http:Request req) {
+        http:Response res = new;
+
+        // Use a util method to set a string payload.
+        res.setPayload("Hello, World!");
+
+        // Send the response back to the caller.
+        var result = caller->respond(res);
+        if (result is error) {
+            log:printError("Error sending response", err = result);
+        }
+    }
+}
+```
+
+Now you can add the Kubernetes annotations that are required to generate the Kubernetes deployment artifacts.
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+import ballerinax/kubernetes;
+
+@kubernetes:Service {
+    name:"hello-world",
+    serviceType:"LoadBalancer",
+    port:80
+}
+listener http:Listener httpListener = new(9090);
+
+@kubernetes:Deployment {
+    enableLiveness:true,
+    image:"<username>/hello_world_service:latest",
+    push:true,
+    username:"<username>",
+    password:"<password"
+}
+// By default, Ballerina exposes a service via HTTP/1.1.
+service hello on httpListener {
+
+    // Invoke all resources with arguments of server connector and request.
+    resource function sayHello(http:Caller caller, http:Request req) {
+        http:Response res = new;
+
+        // Use a util method to set a string payload.
+        res.setPayload("Hello, World!");
+
+        // Send the response back to the caller.
+        var result = caller->respond(res);
+        if (result is error) {
+            log:printError("Error sending response", err = result);
+        }
+    }
+}
+
+```
+We will be building a Docker image here and publishing it to Docker Hub. This is required, since we cannot simply have the Docker image in the local registry, and run the Kubernetes applicates in GKE, where it needs to have access to the Docker image in a globally accessible location. For this, an image name should be given in the format $username/$image_name in the "image" property, and "username" and "password" properties needs to contain the Docker Hub account username and password respectively. The property "push" is set to "true" to signal the build process to push the build Docker image to Docker Hub.
```

Comment:
```suggestion
We will be building a Docker image here and publishing it to Docker Hub. This is required, since we cannot simply have the Docker image in the local registry, and run the Kubernetes applicates in GKE, where it needs to have access to the Docker image in a globally accessible location. For this, an image name should be given in the format `$username/$image_name` in the `image` property, and `username` and `password` properties needs to contain the Docker Hub account username and password respectively. The property "push" is set to "true" to signal the build process to push the build Docker image to Docker Hub.

```